project1/applications.py

```diff
@@ -47,6 +49,38 @@ def result():
     else:
         return render_template("Registration.html")
 
+@app.route("/bookpage", methods=[ 'POST', 'GET'])
+def review_page():
+    db.create_all()
+    if request.method == 'POST':
+        r = reviews(request.form["isbn"], request.form["user"], request.form["rate"], request.form["review"])
+        data2 = reviews.query.filter_by(user = request.form['user']).first()
+        data1= reviews.query.filter_by(isbn = request.form['isbn']).first()
```

Comment:
Please change it to reviews.query.filter_by(user = request.form['user'], isbn = request.form['isbn]).first(). This will check if the user has reviewed for that book or not. In your code what it checks is, Has the user reviewed at least once(not for this book) in data2 and in data1 we are checking if there are any reviews for the book.